### PR TITLE
Organization User List 

### PR DESCRIPTION
**Pull request by poojakarma**
Procedure for Users list of organization 
Step : 1 create functions in service to get organization list and organization user list  
Step : 2 create function in component to get result of API from service
Step : 3 Show Result of organization user list on UI (Table)
Step : 4 Add pagination 
Step :5 Add sorting feature on column
Step :6 Searching for all columns
Step :7 Work on checkbox

Video of Implementation:
https://drive.google.com/file/d/1zMTjUU-wnA7iSogIQE9O3-WHqyyILKB_/view?usp=share_link

### CODE REVIEW

File: src/locales/en/en.json [hunk 1]
```diff
@@ -7,5 +7,13 @@
   "ORGANIZATION":"Organization",
   "ORGANIZATION_NAME":"Organization Name",
   "CHANNEL_NAME":"Channel Name",
-  "CHANNEL_ID":"Channel ID"
+  "CHANNEL_ID":"Channel ID",
+  "FIRST_NAME":"First Name",
```
Comment (manojLondhe) [inline src/locales/en/en.json]: use ENTITY NAME / VIew name as prefix to all newly added lang constants 

eg   FIRST_NAME =>   USER_FIRST_NAME / USER_LIST_FIRST_NAME

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 2]
```diff
@@ -5,24 +5,28 @@
             <p-toolbar styleClass="mb-4">
                 <ng-template pTemplate="left">
                     <div class="my-2">
-                        <button pButton pRipple label="Add User" icon="pi pi-plus" class="p-button-success mr-2" (click)="openNew()"></button>
-                        <button pButton pRipple label="Delete" icon="pi pi-trash" class="p-button-danger" (click)="deleteSelectedUsers()" [disabled]="!selectedUsers || !selectedUsers.length"></button>
+                        <button pButton pRipple label="Add User" icon="pi pi-plus" class="p-button-success mr-2"
```
Comment (manojLondhe) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: need to translate label?

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 3]
```diff
@@ -5,24 +5,28 @@
...
-                <ng-template pTemplate="right">
-                    <p-fileUpload mode="basic" accept="image/*" [maxFileSize]="1000000" label="Import" chooseLabel="Import" class="mr-2 inline-block"></p-fileUpload>
-                    <button pButton pRipple label="Export" icon="pi pi-upload" class="p-button-help" (click)="dt.exportCSV()"></button>
-                </ng-template>
             </p-toolbar>
-
-            <p-table #dt [value]="users" [columns]="cols" responsiveLayout="scroll" [rows]="10" [globalFilterFields]="['name','country.name','representative.name','status']" [rows]="10" [paginator]="true" [rowsPerPageOptions]="[10,20,30]" [showCurrentPageReport]="true" currentPageReportTemplate="Showing {first} to {last} of {totalRecords} entries" [(selection)]="selectedUsers" selectionMode="multiple" [rowHover]="true" dataKey="id">
+            <p-table #myTable [value]="OrganizationsUsersList" [columns]="cols" [loading]='loading'
+                [totalRecords]="OrganizationsUsersList.length" responsiveLayout="scroll" [rows]="10" [paginator]="true"
+                [globalFilterFields]="['rootOrgName','firstName','lastName','email','phone',]"
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: this should come from the config/ts file

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 4]
```diff
@@ -5,24 +5,28 @@
...
+                [rowsPerPageOptions]="[10,20,30]" [showCurrentPageReport]="true"
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: this should come from the config/ts file

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 5]
```diff
@@ -5,24 +5,28 @@
...
                     <div class="flex flex-column md:flex-row md:justify-content-between md:align-items-center">
-                        <h5 class="m-0">Users</h5>
+                        <h5 class="m-0">{{ 'USER' | i18next }}</h5>
+                        <p-message severity="warn" text="{{ 'ORGANIZATION_USER_LIST_WARNING' | i18next }}"
+                               *ngIf="!OrganizationsUsersList || OrganizationsUsersList.length === 0"></p-message>
                         <span class="block mt-2 md:mt-0 p-input-icon-left">
                             <i class="pi pi-search"></i>
-                            <input pInputText type="text" (input)="onGlobalFilter(dt, $event)" placeholder="Search..."  class="w-full sm:w-auto"/>
+                            <input pInputText type="text" (input)="onGlobalFilter(myTable, $event)"
+                                placeholder="Search..." class="w-full sm:w-auto" />
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: Use language const for placeholder also

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html (hunk 4 above)
Comment (gouravmore) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: @poojakarma Can you remove this page number also [rowsPerPageOptions] ="[10,20,30]" from the HTML file

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 6]
```diff
@@ -5,24 +5,28 @@
...
-
-                <ng-template pTemplate="right">
-                    <p-fileUpload mode="basic" accept="image/*" [maxFileSize]="1000000" label="Import" chooseLabel="Import" class="mr-2 inline-block"></p-fileUpload>
-                    <button pButton pRipple label="Export" icon="pi pi-upload" class="p-button-help" (click)="dt.exportCSV()"></button>
-                </ng-template>
             </p-toolbar>
-
-            <p-table #dt [value]="users" [columns]="cols" responsiveLayout="scroll" [rows]="10" [globalFilterFields]="['name','country.name','representative.name','status']" [rows]="10" [paginator]="true" [rowsPerPageOptions]="[10,20,30]" [showCurrentPageReport]="true" currentPageReportTemplate="Showing {first} to {last} of {totalRecords} entries" [(selection)]="selectedUsers" selectionMode="multiple" [rowHover]="true" dataKey="id">
+            <p-table #myTable [value]="OrganizationsUsersList" [columns]="cols" [loading]='loading'
+                [totalRecords]="OrganizationsUsersList.length" responsiveLayout="scroll" [rows]="10" [paginator]="true"
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: @poojakarma Can you remove this page number also [rows]="10" from the HTML file

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html (hunk 4 above)
Comment (poojakarma) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: ok

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html (hunk 6 above)
Comment (poojakarma) [reply src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: ok sir

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 7]
```diff
@@ -47,7 +51,7 @@ <h5 class="m-0">Users</h5>
                             <p-tableCheckbox [value]="user"></p-tableCheckbox>
                         </td>
                         <td style="width:14%; min-width:10rem;"><span class="p-column-title">Organization</span>
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: Remove all the inline css from the HTML files.

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 8]
```diff
@@ -31,13 +35,13 @@ <h5 class="m-0">Users</h5>
                         <th style="width: 3rem">
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: @poojakarma

File: src/app/sb-admin/components/pages/sb-user/sbuser.component.html [hunk 9]
```diff
@@ -46,118 +50,37 @@ <h5 class="m-0">Users</h5>
...
+                        <td >
                             <span class="p-column-title">Last Name</span>
                             {{user.lastName}}
                         </td>
-                        <td style="width:14%; min-width:8rem;">
+                        <td>
                             <span class="p-column-title">Email</span>
                             {{user.email}}
                         </td>
                         <td style="width:14%; min-width:10rem;">
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-user/sbuser.component.html]: @poojakarma remove this inline css

**Review by gouravmore — APPROVED** [review]

File: src/locales/en/en.json [hunk 10]
```diff
@@ -6,6 +6,17 @@
...
   "ORGANIZATION_NAME":"Organization Name",
-  "CHANNEL_NAME":"Channel Name",
-  "CHANNEL_ID":"Channel ID"
+  "ORGANIZATION_CHANNEL_NAME":"Channel Name",
+  "ORGANIZATION_CHANNEL_ID":"Channel ID",
+  "USER_FIRST_NAME":"First Name",
+  "USER_LAST_NAME":"Last Name",
+  "USER_EMAIL":"Email",
+  "USER_PHONE":"Phone",
+  "STATUS":"Status",
```
Comment (manojLondhe) [inline src/locales/en/en.json]: add some prefix to all lang constants

- either screen name eg: ORG_DASHBOARD_ / USER_DASHBOARD_STATUS => `ORG_DASHBOARD_STATUS / USER_DASHBOARD_STATUS`
- or entity name eg: USER_ => `USER_STATUS`
- or these are not entity / screen specific eg: COMMON_ =>`COMMON_STATUS`

**Pull request merged**